modules/prices/src/lib.rs

```diff
@@ -219,3 +217,40 @@ impl<T: Config> PriceProvider<CurrencyId> for Pallet<T> {
 		<Pallet<T>>::deposit_event(Event::UnlockPrice(currency_id));
 	}
 }
+
+/// https://en.wikipedia.org/wiki/Integer_square_root
+fn integer_sqrt(n: U256) -> U256 {
```

Comment:
add TODO to use https://github.com/paritytech/parity-common/pull/554 once it is available